hydroflow/examples/kvs_replicated/server.rs

```diff
@@ -0,0 +1,86 @@
+use std::net::SocketAddr;
+
+use hydroflow::hydroflow_syntax;
+use hydroflow::scheduled::graph::Hydroflow;
+use hydroflow::util::{UdpSink, UdpStream};
+
+use crate::protocol::{KvsMessage, KvsMessageWithAddr};
+use crate::GraphType;
+
+pub(crate) async fn run_server(
+    outbound: UdpSink,
+    inbound: UdpStream,
+    graph: Option<GraphType>,
+    peer_server: Option<SocketAddr>,
+) {
+    println!("Server live!");
+
+    let mut df: Hydroflow = hydroflow_syntax! {
+        // Network channels
+        network_send = union() -> dest_sink_serde(outbound);
+        network_recv = source_stream_serde(inbound)
+            -> _upcast(Some(Delta))
+            -> map(Result::unwrap)
+            -> inspect(|(msg, addr)| println!("Message received {:?} from {:?}", msg, addr))
+            -> map(|(msg, addr)| KvsMessageWithAddr::from_message(msg, addr))
+            -> demux_enum::<KvsMessageWithAddr>();
+        network_recv[ServerResponse] -> for_each(|(key, value, addr)| eprintln!("Unexpected server response {:?}->{:?} from {:?}", key, value, addr));
+        peers = network_recv[PeerJoin] -> map(|(peer_addr,)| peer_addr) -> tee();
+        network_recv[PeerGossip] -> writes;
+        network_recv[ClientPut] -> writes;
+        writes = union() -> tee();
+        gets = network_recv[ClientGet];
+
+        // Join as a peer if peer_server is set.
+        source_iter_delta(peer_server) -> map(|peer_addr| (KvsMessage::PeerJoin, peer_addr)) -> network_send;
+
+        // join PUTs and GETs by key
+        writes -> map(|(key, value, _addr)| (key, value)) -> writes_store;
+        writes_store = persist() -> tee();
+        writes_store -> [0]lookup;
+        gets -> [1]lookup;
+        lookup = join();
```

Comment:
join_multiset?

what happens if you get two get requests in the same tick from the same client